core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -174,6 +174,35 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
     getOption(key).getOrElse(defaultValue)
   }
 
+  /** Get a time parameter as seconds; throws a NoSuchElementException if it's not set. If no 
+    * suffix is provided then seconds are assumed. 
+    */
+  def getTimeAsSec(key: String): Long = {
+    Utils.timeStringAsSec(getOption(key).getOrElse(throw new NoSuchElementException(key)))
```

Comment:
I think you can just call `get(key)` and rely on it to perform exactly this logic? (Am I missing some separate discussion we already had on this?)